xarray/core/variable.py

```diff
@@ -59,12 +58,8 @@
 )
 
 NON_NUMPY_SUPPORTED_ARRAY_TYPES = (
-    (
-        indexing.ExplicitlyIndexed,
-        pd.Index,
-    )
-    + dask_array_type
-    + cupy_array_type
```

Comment:
I was able to remove cupy here and still have `test_cupy.py` pass, but I couldn't remove it in the other two places that cupy is special-cased, because 
1) something goes wrong in `duck_array_ops.asarray`
2) cupy doesn't implement a`.to_numpy` method to convert to numpy, instead needing `.get`.

Reply:
Yeah cupy doesn't like `np.asarray` so we'll have to special-case it. Thanks for checking!